packages/devtools_app/lib/src/shared/table/_table_row.dart

```diff
@@ -494,67 +564,35 @@ class _TableRowState<T> extends State<TableRow<T>>
       );
     }
 
-    final rowDisplayParts = <_TableRowPartDisplayType>[];
-    final groups = widget.columnGroups;
-    if (groups != null && groups.isNotEmpty) {
-      for (int i = 0; i < groups.length; i++) {
-        final groupParts = List.generate(
-          groups[i].range.size as int,
-          (index) => _TableRowPartDisplayType.column,
-        ).joinWith(_TableRowPartDisplayType.columnSpacer);
-        rowDisplayParts.addAll(groupParts);
-        if (i < groups.length - 1) {
-          rowDisplayParts.add(_TableRowPartDisplayType.columnGroupSpacer);
-        }
-      }
-    } else {
-      final parts = List.generate(
-        widget.columns.length,
-        (_) => _TableRowPartDisplayType.column,
-      ).joinWith(_TableRowPartDisplayType.columnSpacer);
-      rowDisplayParts.addAll(parts);
-    }
-
-    // Maps the indices from [rowDisplayParts] to the corresponding index of
-    // each column in [widget.columns].
-    final columnIndexMap = <int, int>{};
-    // Add scope to guarantee [columnIndexTracker] is not used outside of this
-    // block.
-    {
-      var columnIndexTracker = 0;
-      for (int i = 0; i < rowDisplayParts.length; i++) {
-        final type = rowDisplayParts[i];
-        if (type == _TableRowPartDisplayType.column) {
-          columnIndexMap[i] = columnIndexTracker;
-          columnIndexTracker++;
-        }
-      }
-    }
-
     Widget rowContent = Padding(
       padding: const EdgeInsets.symmetric(horizontal: defaultSpacing),
-      child: ListView.builder(
+      child: ExtentDelegateListView(
         scrollDirection: Axis.horizontal,
+        physics: const ClampingScrollPhysics(),
         controller: scrollController,
-        itemCount: widget.columns.length + widget.columns.numSpacers,
-        itemBuilder: (context, int i) {
-          final displayTypeForIndex = rowDisplayParts[i];
-          switch (displayTypeForIndex) {
-            case _TableRowPartDisplayType.column:
-              final index = columnIndexMap[i]!;
-              return columnFor(
-                widget.columns[index],
-                widget.columnWidths[index],
-              );
-            case _TableRowPartDisplayType.columnSpacer:
-              return const SizedBox(
-                width: columnSpacing,
-                child: VerticalDivider(width: columnSpacing),
-              );
-            case _TableRowPartDisplayType.columnGroupSpacer:
-              return const _ColumnGroupSpacer();
-          }
-        },
+        extentDelegate: rowExtentDelegate,
+        childrenDelegate: SliverChildBuilderDelegate(
+          (context, int i) {
+            final columnIndexMap = _columnIndexMapHelper(_rowDisplayParts);
```

Comment:
Does this need to be regenerated for each child?

Reply:
maybe not but I'll follow up with another PR looking into that